app/updateSNAPP.ts

```diff
@@ -0,0 +1,135 @@
+        application.packageName}.app`;
+            break;
+        }
+        case WINDOWS: {
+            applicationExecutable = path.join(
+                `${application.packageName || application.name}`,
+                `${application.name || application.packageName}.exe`
+            );
+            break;
+        }
+        case LINUX: {
+            applicationExecutable = `${application.packageName ||
+        application.name}.AppImage`;
+            break;
+            // electron-react-boilerplate-0.1.0-x86_64.AppImage
+        }
+        default: {
+            logger.error( 'Unsupported platform for desktop applications:', platform );
+        }
+    }
+    return applicationExecutable;
+};
+
+const getInstalledLocation = ( application ): string => {
+    const applicationExecutable = getApplicationExecutable( application );
+    const installedPath = path.join( getAppInstallDir(), applicationExecutable );
+
+    logger.error( 'installedPath', installedPath );
+
+    return installedPath;
+};
+
+export const updateSNAPP = ( arg ) => {
+    // We could pass the app object as an arg also if needed
+    // To enable more cross app communication
+    logger.error( 'updateSNAPP called with arg', arg );
+    const application = {
+        name: 'SAFE Network App',
+        packageName: 'safe-network-app'
+    };
+
+    const appLocation = getInstalledLocation( application );
+    let command = appLocation;
+
+    const newEnvironment = {
+        ...process.env,
+        NODE_ENV: 'prod',
+        HOT: 'false'
+    };
+
+    // needs to be actually deleted.
+    delete newEnvironment.HOT;
+
+    logger.warn( 'Opening app via path: ', command );
+
+    if ( platform === MAC_OS ) {
+        command = `open "${command}" ${arg}`;
+
+        exec( command, {
+            // eslint-disable-next-line unicorn/prevent-abbreviations
+            env: newEnvironment
+        } );
+    }
+    if ( platform === WINDOWS ) {
+        execFile( command, [...arg, '--debug'], {
+            // eslint-disable-next-line unicorn/prevent-abbreviations
+            env: newEnvironment
+        } );
+        return;
+    }
+    if ( platform === LINUX ) {
+        logger.warn( 'Opening on linux via spawn command: ', command );
+        // exec on linux doesnt give us a new process, so closing SNAPP
+        // will close the spawned app :|
+        spawn( command, [...arg, '--debug'], {
+            // eslint-disable-next-line unicorn/prevent-abbreviations
+            env: newEnvironment,
+            detached: true
+        } );
+    }
+};
+
+export const checkIfAppIsRunning = async () => {
+    // We could pass the app object as an arg also if needed
+    // To enable more cross app communication
+    const application = {
+        name: 'SAFE Network App',
+        packageName: 'safe-network-app'
+    };
+    const appList = await psList();
+    const appIsRunning = appList.find(
+        ( element ) => element.name === `${application.name}.exe`
```

Comment:
how would this work on mac/linux?

Reply:
The package works on all three platforms and returns a Promise<Array> with the running processes.

Reply:
specifically `${application.name}.exe` ?

Reply:
In this case it would Look for `SAFE Network App.exe` to check if a process is running and resturn an array of the current SNAPP running processes